src/index.ts

```diff
@@ -0,0 +1,103 @@
+import * as Bluebird from 'bluebird';
+import { PinejsClientCoreFactory } from 'pinejs-client-core';
+import type * as express from 'express';
+import type { CallbackHandler, Response, Test } from 'supertest';
+import supertest from './supertest';
+import { expect } from './chai';
+import type { UserParam, Overwrite } from './common';
+
+type supportedMethod = 'get' | 'put' | 'patch' | 'post' | 'delete';
+
+// Use `T = any` by default, just like supertest does, but also support specifying
+// the expected type, so that we can avoid the ugly `() as Something` castings.
+type PromiseResult<T = any> = Omit<Test, 'expect' | 'then'> &
+	Promise<Overwrite<Response, { body: T }>> & {
+		// Atm we can't override just the return type of a method w/o re-defining all its variants.
+		expect(
+			statusOrCheckerOrBody:
+				| number
+				| ((res: Response) => any)
+				| string
+				| RegExp
+				| object,
+			callback?: CallbackHandler,
+		): PromiseResult<T>;
+		expect(
+			status: number,
+			body: any,
+			callback?: CallbackHandler,
+		): PromiseResult<T>;
+		expect(
+			field: string,
+			val: string | RegExp,
+			callback?: CallbackHandler,
+		): PromiseResult<T>;
+	};
+
+interface BackendParams {
+	app: express.Express;
+}
+
+/** A pine testing client fused with the api of supertest */
+export class PineTest extends PinejsClientCoreFactory(Bluebird)<
+	PineTest,
+	PromiseResult,
+	PromiseResult
+> {
+	constructor(
+		params: PinejsClientCoreFactory.Params,
+		public backendParams: BackendParams,
+	) {
+		super(params);
+	}
+
+	public get<T = any>(
+		params: PinejsClientCoreFactory.Params,
+	): PromiseResult<T> {
+		// Use a different const, since if we just re-assign `params`
+		// inside the `expect(() => {})` TS forgets that it's ensured
+		// to be a ParamsObj and will complain.
+		const normalizedParams =
+			typeof params === 'string' ? { url: params } : params;
+		normalizedParams.method = 'GET';
+
+		return this.request(normalizedParams).expect((response) => {
+			const { error, body } = response;
+			if (error) {
+				return;
+			}
+
+			expect(() => {
+				response.body = this.transformGetResult(normalizedParams)(body);
+			}).to.not.throw();
+		});
+	}
+
+	public post<T = any>(
+		params: PinejsClientCoreFactory.Params,
+	): PromiseResult<T> {
+		return super.post(params);
+	}
+
+	_request({
```

Comment:
Is this overriding a method from PinejsClientCoreFactory? It's prefixed with an underscore, which I take it means a private method, so it would be nice to have a comment briefly describing why the override is needed.

Reply:
@dfunckt It's mandatory to define _request since it's abstract in the base class (PinejsClientCoreFactory), so I didn't add any comment to describe the override part.
See: https://github.com/balena-io-modules/pinejs-client-js/blob/310a1799bf0612f31324213c2367229bbf2d44b6/core.ts#L1279-L1298

Essentially all other methods are overrides :)

I'm not sure about the naming convention, it could have been requestBase I guess, to avoid confusing it with a private one. I guess that we just never cared to do a breaking change just for the name of this function.
See: https://github.com/balena-io-modules/pinejs-client-request/blob/4189de2ce31c161bb29903d8344e34525e102a33/request.ts#L63